game-core/src/main/java/games/strategy/triplea/delegate/battle/BattleState.java

```diff
@@ -90,7 +90,10 @@ public boolean isFirstRound() {
 
   void retreatUnits(Side side, Collection<Unit> units);
 
-  Collection<Unit> removeNonCombatants(Side side);
+  /**
+   * @param removeAaUnitsThatFireThisRound Should AA units that fire this round be removed or not
+   */
+  Collection<Unit> removeNonCombatants(Side side, boolean removeAaUnitsThatFireThisRound);
```

Comment:
Rather than having battle state be told whether to remove AA units, it seems like AA units should know that. Perhaps after a unit fires, there could be a check run on that unit type to determine if the unit should be removed after firing. 

I mention that for a few reasons:
- boolean parameters are best avoided (automatic SRP violations, usually violate encapsulation and DRY)
- In this case it does seem like we have a DRY violation, the 'removeNonCombatants' method is not able to fully own that logic and needs to be told how to properly remove non-combatants, it is unable to determine which units are non-combatants on its own.

Thoughts, comments?

Reply:
Currently, units don't know that they fired or are even dead (even the amount of hits that a unit has doesn't indicate if it is dead or not).  So, the idea is great but it is not easily doable at this time.